### ISSUE TITLE
Error in reading cif

### ISSUE THREAD

**Issue by hepesu** [DanPorter/Dans_Diffraction#1]
Hi, I am going to use this great project to simulate X-ray diffraction result from cif. But when I , tried, I met an error in reading cif downloaded from CCDC, the entry id is 1000175. It gave this message.

> Traceback (most recent call last):
  File "D:/Workspace/ccdc/sims/test.py", line 3, in <module>
    xtl = Crystal('1000175.cif')
  File "D:\Workspace\ccdc\sims\Dans_Diffraction\classes_crystal.py", line 99, in __init__
    cifvals = fc.readcif(filename)
  File "D:\Workspace\ccdc\sims\Dans_Diffraction\functions_crystallography.py", line 89, in readcif
    while lines[n].split()[0][0] == '_':
IndexError: list index out of range

I tried to fix this by deleting lines that made the error, but it did not work.

I have a suggestion for this, how about using existing cif parser for cif parsing?

Thank you!

**Comment by DanPorter** [comment]
Hi again,

I've just pushed an update to a few files in Dans_Diffraction that should solve the problem. If you pull the latest version it should now be fixed.

The problem was actually the cif file - line 67 was blank after __symmetry_equiv_pos_as_xyz_, which is not usual and resulted in the error. 

I have now updated the function to remove any blank lines at the start.

Do let me know if you have any other problems!

P.S. Purely out of curiosity - how did you find my code and what are you hoping to use it for?

**Comment by hepesu** [comment]
Hi Dan,

The new code works. It is great! Thank you!

Maybe pycifrw is a good choice.

I am working on a project which aims solving structure directly by deep learning model. The raw input is hkl data. To enrich the dataset, I decide to generate hkl data using cif without hkl data embedded. Some softwares can do the job, but making them work with python is a problem. So I did the search on github, and I found your great project.

As for usage detail, firstly I am going to use your project to generate hkl data.Then I will write some code based on your code, which uses tensorflow instead of numpy to meet the demand of loss function for learning.

Thank you again.

Hepesu
